frame/support/src/inherent.rs

```diff
@@ -44,15 +53,17 @@ pub trait ProvideInherent {
 	/// implementation returns this.
 	///
 	/// - `Ok(Some(e))` indicates that this inherent is required in this block. `construct_runtime!`
-	/// will call this function from in its implementation of `fn check_extrinsics`.
+	/// will call this function in its implementation of `fn check_extrinsics`.
 	/// If the inherent is not present, it will return `e`.
 	///
 	/// - `Err(_)` indicates that this function failed and further operations should be aborted.
 	///
-	/// NOTE: If inherent is required then the runtime asserts that the block contains at least
+	/// NOTE: If the inherent is required then the runtime asserts that the block contains at least
 	/// one inherent for which:
 	/// * type is [`Self::Call`],
 	/// * [`Self::is_inherent`] returns true.
+	///
+	/// NOTE: This is currently only checked by block producers, not all full nodes.
```

Comment:
That is not true. 

`check_inherents` is currently called by all nodes. The idea is that it is only required by validators. However, that isn't really important here.

Reply:
So reword to "this is currently checked by all full nodes but only guaranteed to be checked by validators" or drop entirely?
IIRC the intention behind this was to give people guidance when using inherents to avoid insecurity.